src/main/java/io/stargate/sgv2/jsonapi/service/resolver/matcher/TableFilterResolver.java

```diff
@@ -75,20 +90,44 @@ private static List<DBFilterBase> findDynamic(CaptureExpression captureExpressio
 
     // TODO: How do we know what the CmdT of the JsonLiteral<CmdT> from .value() is ?
     for (FilterOperation<?> filterOperation : captureExpression.filterOperations()) {
+      final Object rhsValue = filterOperation.operand().value();
       if (captureExpression.marker() == DYNAMIC_TEXT_GROUP) {
         filters.add(
             new TextTableFilter(
                 captureExpression.path(),
                 NativeTypeTableFilter.Operator.from(
                     (ValueComparisonOperator) filterOperation.operator()),
-                (String) filterOperation.operand().value()));
+                (String) rhsValue));
       } else if (captureExpression.marker() == DYNAMIC_NUMBER_GROUP) {
         filters.add(
             new NumberTableFilter(
                 captureExpression.path(),
                 NativeTypeTableFilter.Operator.from(
                     (ValueComparisonOperator) filterOperation.operator()),
-                (BigDecimal) filterOperation.operand().value()));
+                (BigDecimal) rhsValue));
+      } else if (captureExpression.marker() == DYNAMIC_DOCID_GROUP) {
+        Object actualValue = ((DocumentId) rhsValue).value();
```

Comment:
Should probably be refactored to avoid duplicating code to create Text/NumberTableFilter.